sdk/src/crypto/asn1/rfc5652.rs

```diff
@@ -20,15 +20,190 @@ use std::{
 };
 
 use bcder::{
-    decode::{Constructed, DecodeError, Source},
+    decode::{Constructed, DecodeError, Primitive, Source},
     encode,
     encode::{PrimitiveContent, Values},
     BitString, Captured, ConstOid, Integer, Mode, OctetString, Oid, Tag,
 };
-use x509_certificate::{asn1time::*, rfc3280::*, rfc5280::*, rfc5652::*};
+use der::Decode;
 
+// Re-use common types from parent module
+use super::{AlgorithmIdentifier, GeneralizedTime};
 use crate::crypto::asn1::rfc3281::AttributeCertificate;
 
+// Define local bcder-compatible types to replace x509_certificate imports
+
+/// UTC time compatible with bcder - hybrid approach using der crate
+#[derive(Clone, Debug)]
+pub struct UtcTime {
+    // Store the der type for proper time parsing (handles Y2K correctly)
+    der_time: der::asn1::UtcTime,
+    // Cache the DER encoding for bcder compatibility
+    der_bytes: Vec<u8>,
+}
+
+impl PartialEq for UtcTime {
+    fn eq(&self, other: &Self) -> bool {
+        self.der_time == other.der_time
+    }
+}
+
+impl Eq for UtcTime {}
+
+impl UtcTime {
+    pub fn from_primitive<S: Source>(
+        prim: &mut Primitive<S>,
+    ) -> Result<Self, DecodeError<S::Error>> {
+        // Get the raw bytes (without tag/length)
+        let bytes = prim.take_all()?;
+        // Reconstruct DER encoding using helper (tag 0x17 = UTC_TIME)
+        let der_bytes = super::reconstruct_der_bytes(0x17, bytes.as_ref());
+        // Parse with der crate - it handles Y2K correctly and validates format
+        let der_time = der::asn1::UtcTime::from_der(&der_bytes)
+            .map_err(|_| DecodeError::content("invalid UtcTime", prim.pos()))?;
+        Ok(UtcTime {
+            der_time,
+            der_bytes,
+        })
+    }
+
+    /// Convert to chrono DateTime
+    pub fn to_datetime(&self) -> chrono::DateTime<chrono::Utc> {
+        // Convert der's time to SystemTime, then to chrono
+        let system_time = self.der_time.to_system_time();
+
+        // Most platforms (including wasm32-wasi) support SystemTime conversion directly
+        #[cfg(any(not(target_arch = "wasm32"), target_os = "wasi"))]
+        {
+            system_time.into()
+        }
+
+        // Only wasm32-unknown-unknown needs to go through timestamp
+        #[cfg(all(target_arch = "wasm32", not(target_os = "wasi")))]
+        {
+            use chrono::TimeZone;
+            let duration = system_time
+                .duration_since(std::time::UNIX_EPOCH)
+                .unwrap_or(std::time::Duration::from_secs(0));
+            chrono::Utc
+                .timestamp_opt(duration.as_secs() as i64, duration.subsec_nanos())
+                .single()
+                .unwrap_or_else(|| chrono::Utc.timestamp_opt(0, 0).unwrap())
+        }
+    }
```

Comment:
Are there any tests to make sure you got this right?  This stuff is easy to mess up.